data-service/src/main/java/uk/gov/gchq/palisade/service/data/exception/ErrorDetails.java

```diff
@@ -46,78 +47,80 @@ public ErrorDetails(final ZonedDateTime date, final String message, final String
         this.stackTrace = Arrays.asList(trace);
     }
 
-    public void setLocalDate(final ZonedDateTime date) {
-        requireNonNull(date, "Date cannot be null");
+    @Generated
+    public ZonedDateTime getDate() {
+        return date;
+    }
+
+    @Generated
+    public void setDate(final ZonedDateTime date) {
+        requireNonNull(date);
         this.date = date;
     }
 
-    public void setDate(final String dateString) {
-        requireNonNull(dateString, "String value cannot be null");
-        this.date = ZonedDateTime.parse(dateString);
+    @Generated
+    public String getMessage() {
+        return message;
     }
 
+    @Generated
     public void setMessage(final String message) {
-        requireNonNull(message, "Message cannot be null");
+        requireNonNull(message);
```

Comment:
some of the requireNonNulls have a message, some dont, can this be standardised?

Reply:
See above comment